### ISSUE TITLE
Surface syncgroup changes in watch stream

### ISSUE THREAD

**Issue by asadovsky** [vanadium/issues#764]
(splitting an old issue into two separate parts)
+cc @rdvg @hpucha @rogulenko 

Syncbase apps often consume the watch stream to find out about changes and re-render the UI. Currently, syncgroup changes are not surfaced in the watch stream, so UIs will not update to reflect the latest remote changes to a syncgroup ACL (for example). We should surface syncgroup changes in the watch stream.

**Comment by ivanpi-mtv** [comment]
Any interest in watching for data ACL changes as well?

**Comment by asadovsky** [comment]
Yes. I guess I didn't mention this in the issue description, but we should definitely surface those as well.

**Comment by asadovsky** [comment]
I'll take this on. Also +cc @aghassemi
